### PR TITLE
Feature/setup testing

### PR DESCRIPTION
**Pull request by CamilleFranceschi**

### CODE REVIEW

**Comment by RonnieOnRails** [comment]
y a encore un commit merge xD

File: app/models/card.rb [hunk 1]
```diff
@@ -3,15 +3,16 @@
 class Card < ApplicationRecord
   belongs_to :brand
   belongs_to :company
-  # check unicty of a card related to a brand for one company
+  # check unicity of a card related to a brand for one company
   validates :brand_id, uniqueness: { scope: :company_id }
   has_many :point_transactions
 
   # REDIS LOCK !!!!
   def synchronize_cached_points(points, sign)
```
Comment (MathieuDerelle) [inline app/models/card.rb]: points & sign useless

File: app/models/card.rb [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
-    # TODO:  pas all mais ceux de la company
-    self.cached_available_points = PositivePointTransaction.all.sum(&:points) - NegativePointTransaction.all.sum(&:points)
-    self.cached_total_points = PositivePointTransaction.all.sum(&:points)
+    ppt = PositivePointTransaction.where(card: self).sum(&:points)
+    npt = NegativePointTransaction.where(card: self).sum(&:points)
+    self.cached_available_points = ppt - npt
```
Comment (gaugendre) [inline app/models/card.rb]: = point_transactions.sum(&points)

File: app/models/card.rb [hunk 3]
```diff
@@ -3,15 +3,16 @@
...
+    self.cached_total_points = ppt
```
Comment (gaugendre) [inline app/models/card.rb]: tu peux creer has_many :positive_point_transactions, class_name: PositivePointTransaction.to_s si tu veux

File: app/models/card.rb (hunk 2 above)
Comment (MathieuDerelle) [reply app/models/card.rb]: `point_transactions.sum('points * sign')` alors

Comment (gaugendre) [reply app/models/card.rb]: oui

**Pull request merged**